Adaptaciones para vista de presidente

gianpietro1 opened the pull request:
Cuando hay presidentes:
- Remueve número
- Indica "Candidata(o) a presidencia" en lugar de "postula por..."

gianpietro1 committed:
Basic adaptations for presidents
- Remove number
- Replace district for "Candidata(o) a presidencia)"

wixo commented on components/CandidateBigCard/index.js:
```diff
@@ -17,6 +17,16 @@ import capitalize from '../Steps/PartyResults/startCasePeruvianRegions';
 
 const profileIconBaseUrl = 'https://declara.jne.gob.pe/Assets/Fotos-HojaVida/';
 
+const postulaContent = (role, district, genre) => {
```
Es `postula` la abreviación de algo? de repente podríamos usar el nombre `runType`

gianpietro1 replied:
@wixo lo cambié por contentPerCandidateType

wixo replied:
Okay, va!

wixo commented on components/CandidateBigCard/index.js:
```diff
@@ -17,6 +17,16 @@ import capitalize from '../Steps/PartyResults/startCasePeruvianRegions';
 const profileIconBaseUrl = 'https://declara.jne.gob.pe/Assets/Fotos-HojaVida/';
 
+const postulaContent = (role, district, genre) => {
+  if (role.indexOf('CONGRESISTA') > -1) {
```
Podemos reemplazar `indexOf` con `Array.includes`? Lo haría más fácil de leer.

gianpietro1 replied:
reemplazado! gracias.

wixo commented on components/CandidateBigCard/index.js:
```diff
@@ -17,6 +17,16 @@ import capitalize from '../Steps/PartyResults/startCasePeruvianRegions';
+    return district !== 'PERUANOS RESIDENTES EN EL EXTRANJERO'
+      ? `Postula por ${capitalize(district)}`
+      : `Postula al Extranjero`;
+  } else {
```
Los `else` son difíciles de leer, ya que estamos retornando en la línea 22, aquí podríamos escribir otro if con el otro caso, que podría ser la negación de lo que escribamos en la línea 21.

gianpietro1 replied:
reemplazado! de paso implementé el soporte de vicepresidentes "puros", explico:
- el primer IF hace match de "congresistas" y "vicepresidentes-congresistas" (en otro PR subiré la alerta correspondiente al segundo caso)
- el segundo IF hace match de "presidentes" y "vicepresidentes puros", donde aplica el análisis de F o M para poner candidata o candidato, además de especificar si es 1er o 2do VP.

wixo commented on components/CandidateBigCard/index.js:
```diff
@@ -17,6 +17,16 @@ import capitalize from '../Steps/PartyResults/startCasePeruvianRegions';
+      ? `Postula por ${capitalize(district)}`
+      : `Postula al Extranjero`;
+  } else {
+    return genre === 'F' ? `Candidata a Presidente` : `Candidato a Presidente`;
```
De repente eslint ahora te va a indicar que los backticks se prefieren usar cuando hay interpolación dentro del string, en este caso solo podríamos usar `'Candidata a Presidente'` por ejemplo.

gianpietro1 replied:
Dejé los backticks pero porque ahora sí estamos interpolando.

wixo reviewed (changes requested):
Feedback temprano, gracias @gianpietro1!

gianpietro1 committed:
candidate twitter v1

gianpietro1 committed:
support vicepresidents & minor optimizations

gianpietro1 committed:
candidate facebook v1

gianpietro1 commented:
@wixo éste ya está para revisión final, gracias

wixo commented on components/CandidateBigCard/index.js:
```diff
@@ -12,11 +12,33 @@ import {
+      ? '2do Vicepresidente'
+      : 'Presidente';
+    return genre === 'F'
+      ? `Candidata a ${specificRole}`
```
Esta lógica no contempla el caso "Candidata a presidenta". En el caso de Vicepresidente, no suena mejor "Candidata a 1ra vicepresidencia"? he visto que no está en el diseño por eso pregunto.

gianpietro1 replied:
Cierto, no está en el diseño pero creo que puede ser como dices y extenderlo a Presidentes?
- Candidata(o) a Presidencia
- Candidata(o) a 1ra/2da Vicepresidencia

gianpietro1 replied:
por ahora lo puse así en el penúltimo commit

wixo replied:
De acuerdo.

wixo commented on components/CandidateBigCard/index.js:
```diff
@@ -49,28 +71,53 @@ const CandidateBigCard = (props) => {
           <Row>
             <PartyIcon src={partyIconSource(props.candidateParty)} />
-            <NumberIcon>{props.candidateNumber || '5'}</NumberIcon>
+            {props.candidateRole.indexOf('CONGRESISTA') !== -1 ? (
```
Podemos usar `Array.includes` aquí también?

gianpietro1 replied:
oops! va

wixo commented on components/CandidateBigCard/styles.js:
```diff
@@ -41,10 +41,10 @@ export const NumberIcon = styled('span')`
-  width: 71,
-}))``;
+export const ProfileIcon = styled('img')`
+  height: 100px;
```
Por qué este cambio? leí el commit y no encontré una rpta.

gianpietro1 replied:
olvidé hacer el amend para explicar; lo que pasa es que añadir la sección opcional de redes sociales introdujo un height variable al card, y reveló que las fotos variaban de tamaño con la implementación anterior (no investigué por qué), así que cambié a esta alternativa; me avisas cómo proceder

wixo replied:
okay.. puedo hacer pull y probar o haces push y probamos en línea jajaj, cuál prefieres? tienes un url directo para verlo?

gianpietro1 replied:
en el siguiente commit revierto y lo vemos?

gianpietro1 replied:
revertí en el último commit para que veas el problema, compara la altura de las fotos:
https://votupe-ddgne56mi-openpolitica.vercel.app/candidate/134923
https://votupe-ddgne56mi-openpolitica.vercel.app/candidate/134798

wixo replied:
Ah ya, pensé que tenía algo que ver con al adición de los íconos de redes sociales, mi mejor guess es que al ser un layout flex, dependiendo de la longitud del nombre está adecuándose, podemos intentar controlar eso con `flex-shrink` y desde `next/image` con la prop `fill`: https://nextjs.org/docs/api-reference/next/image#layout. Incluso ahí dice que usando `fixed` hace mimic de `img`. Hay buenas optimizaciones con el uso de `next/image` así que sí haría el intento de mantenerlo.

gianpietro1 replied:
@wixo en el commit nuevo dejamos `next/image` y usamos un div alrededor de la imagen para lidiar con el problema de foto variable, que al final tenía que ver con el ancho variable según el nombre del candidato. Me avisas qué te parece, por lo demás estaría listo para merge.

gianpietro1 replied:
(El commit que añade esto es `c945792`, están todos a la misma fecha porque hice un force-lease para rebasear)

wixo commented on components/CandidateBigCard/index.js:
```diff
@@ -49,28 +71,53 @@ const CandidateBigCard = (props) => {
+        <RowBorderUp>
+          <Subtitle>REDES SOCIALES</Subtitle>
+          {props.candidateTwitterLink ? (
+            <SocialLink href={props.candidateTwitterLink} target="_blank">
```
No sé si `eslint` nos previene de esto out of the box (cc: @LuighiV), pero recuerdo que al tener un target="_blank" se recomienda tener un rel noreferrer, más info: https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/

gianpietro1 replied:
sería así? rel="noopener noreferrer"

gianpietro1 replied:
lo puse así por ahora en el penúltimo commit

wixo replied:
pienso que sí 👍

wixo reviewed (changes requested):
Unas preguntas y recomendaciones, gracias @gianpietro1!

wixo commented on components/CandidateBigCard/styles.js:
```diff
@@ -119,7 +122,23 @@ export const Subtitle = styled('h3')`
 `;
+
+export const SocialTwitterIcon = styled('img')`
+  height: 16px;
```
Una más perdón, podemos usar `rems` en vez de `px`?

gianpietro1 replied:
fixed!

gianpietro1 committed:
code & gender optimizations
- Changes presidents/vicepresidents subtitle to neutral gender
- Changes last indexOf for includes
- Translates pixels to rems
- Adds rel="noopener noreferrer" to target _blanks

gianpietro1 committed:
temporary revert ProfileIcon for testing
to test, compare:
http://localhost:3000/candidate/134798
http://localhost:3000/candidate/134923

gianpietro1 committed:
sorrounding profileIcon with div
this keeps next/image and avoids variable image sizes

wixo commented on components/CandidateBigCard/styles.js:
```diff
@@ -119,7 +127,23 @@ export const Subtitle = styled('h3')`
+
+export const SocialLink = styled('a')`
+  text-decoration: none;
+)`;
```
Me parece que ese `)` no debería ir ahí.

gianpietro1 replied:
oops! gracias @wixo ...todo listo 👍

wixo reviewed (changes requested):
Gracias @gianpietro1! casi todo listo, solo ese `)` que no sé como llegó ahí jaja.. también haríamos squash and merge de este PR, estás de acuerdo?

gianpietro1 committed:
remove accidental ")"

wixo reviewed (approved):
gracias @gianpietro1! 🚀
